### PR TITLE
Strong password message

### LINKED ISSUE #7330: Include the password strength criteria in the iOS UI
## 🎯 What needs to be done?
Inform the user of our password strength rules. If they try to signup with a password that doesn't match these rules, inform them even more [in red] 😃  

## 📝 Detailed explanation
Design links an UX: https://www.notion.so/toggl/UX-documentation-for-password-strenght-project-done-9779bca9938b46de8d1c6761b9f30bd3#89d2f5ae71a5444cbce72910a2e1579b

tldr:
- display the "Include 8 or more characters, with at least one uppercase letter, lowercase letter, and a number." below password field
- make sure the copy is always visible when the password field is active
- if user tries to signup with a password that doesn't match the criteria, make the copy red
- the copy turns white again, as soon as they type in a strong password (no need to actually press 'signup' again)

## 🔗 Dependencies
#7329 

## 🏆 Outcome
- The copy is displayed below the password field
- It is always visible when the password field is active
- If the user tries to signup with a password that doesn't match the rules, make the copy red
- The copy turns white again, as soon as they type in a strong password (no need to actually press 'signup' again)

## 👕 Estimated complexity
S

### PR DESCRIPTION
**Pull request by Odrakir**
## What's this?
Changes the way the password error looks according to the issue.

### Relationships
Closes #7330

## How is it done?
Some observable magic. Not much to add.

## :squid: Permissions
Go!

### CODE REVIEW

File: Toggl.iOS/Extensions/Reactive/UIControlExtensions.cs [hunk 1]
```diff
@@ -24,5 +24,17 @@ void changed(object sender, EventArgs args)
...
+            => Observable
+                .FromEventPattern(handler => reactive.Base.EditingDidBegin += handler,
+                    handler => reactive.Base.EditingDidBegin -= handler)
+                .Select(_ => Unit.Default);
+
+        public static IObservable<Unit> EditEnded(this IReactive<UIControl> reactive)
+            => Observable
+                .FromEventPattern(handler => reactive.Base.EditingDidEnd += handler,
+                    handler => reactive.Base.EditingDidEnd -= handler)
+                .Select(_ => Unit.Default);
```
Comment (pdrj) [inline Toggl.iOS/Extensions/Reactive/UIControlExtensions.cs:38]: We don't need these anymore, right?

File: Toggl.iOS/ViewControllers/SignUpViewController.cs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ﻿using System;
 using System.Reactive.Linq;
 using Foundation;
+using GameKit;
```
Comment (pdrj) [inline Toggl.iOS/ViewControllers/SignUpViewController.cs:4]: 🔥

File: Toggl.iOS/ViewControllers/SignUpViewController.cs [hunk 3]
```diff
@@ -79,13 +83,29 @@ public override void ViewDidLoad()
...
+                .DisposedBy(DisposeBag);
+
             //Errors
             ViewModel.EmailError
                 .Subscribe(EmailErrorLabel.Rx().Text())
                 .DisposedBy(DisposeBag);
 
             ViewModel.PasswordError
-                .Subscribe(PasswordErrorLabel.Rx().Text())
+                .Subscribe( error =>
```
Comment (pdrj) [inline Toggl.iOS/ViewControllers/SignUpViewController.cs:104]: The space can be removed I think.

File: Toggl.iOS/Resources/Colors.xcassets/ErrorText.colorset/Contents.json [hunk 4]
```diff
@@ -1,38 +1,38 @@
...
+      ],
+      "color" : {
+        "color-space" : "srgb",
+        "components" : {
+          "alpha" : "1.000",
+          "blue" : "0.020",
+          "green" : "0.020",
+          "red" : "0.886"
+        }
+      },
```
Comment (pdrj) [inline Toggl.iOS/Resources/Colors.xcassets/ErrorText.colorset/Contents.json:30]: Why the changes to this file? Won't this break the error messages in dark mode in the edit view? Maybe we can just use the red that's not aware of dark theme, that will be introduced in #7392?

File: Toggl.iOS/Extensions/Reactive/UIControlExtensions.cs (hunk 1 above)
Comment (Odrakir) [reply Toggl.iOS/Extensions/Reactive/UIControlExtensions.cs:38]: They are nice to have... but ok, I'll delete them

File: Toggl.iOS/ViewControllers/SignUpViewController.cs (hunk 2 above)
Comment (Odrakir) [reply Toggl.iOS/ViewControllers/SignUpViewController.cs:4]: :wat: ?

File: Toggl.iOS/Resources/Colors.xcassets/ErrorText.colorset/Contents.json (hunk 4 above)
Comment (Odrakir) [reply Toggl.iOS/Resources/Colors.xcassets/ErrorText.colorset/Contents.json:30]: The dark mode version for that red color was some sort of white. I assumed someone forgot to add it. I just checked and I think I'm right, errors on the login screen are showing in white when in dark mode for the version in the AppStore.

I can't be 100% sure this isn't breaking something else, but I'm almost sure.

Comment (pdrj) [reply Toggl.iOS/Resources/Colors.xcassets/ErrorText.colorset/Contents.json:30]: It is breaking something else, that's why I introduced that red that is not changning in dark mode ;) 
Screenshot 2020-05-14 at 17 40 25

Comment (Odrakir) [reply Toggl.iOS/Resources/Colors.xcassets/ErrorText.colorset/Contents.json:30]: 🤦 Ok

Comment (Odrakir) [reply Toggl.iOS/Resources/Colors.xcassets/ErrorText.colorset/Contents.json:30]: To be fair, naming that color "ErrorText" doesn't seem like a great idea.

File: Toggl.iOS/Extensions/ColorAssets.cs [hunk 5]
```diff
@@ -22,6 +22,8 @@ public class ColorAssets
         public static readonly UIColor CustomGray5 = UIColor.FromName("CustomGray5");
         public static readonly UIColor CustomGray6 = UIColor.FromName("CustomGray6");
         public static readonly UIColor Spider = UIColor.FromName("Spider");
+        public static readonly UIColor ErrorText = UIColor.FromName("ErrorText");
+        public static readonly UIColor ErrorRed = UIColor.FromName("ErrorRed");
```
Comment (sofokli1) [inline Toggl.iOS/Extensions/ColorAssets.cs:26]: these names are not super clear
